tests/interpreters/test_comparisons.py

```diff
@@ -0,0 +1,44 @@
+from xdsl.interpreters.comparisons import (
+    signed_lower_bound,
+    signed_upper_bound,
+    to_signed,
+    to_unsigned,
+    unsigned_upper_bound,
+)
+
+BITWIDTH = 2
+UNSIGNED_UPPER_BOUND = unsigned_upper_bound(BITWIDTH)
+SIGNED_UPPER_BOUND = signed_upper_bound(BITWIDTH)
+SIGNED_LOWER_BOUND = signed_lower_bound(BITWIDTH)
+
+UNSIGNED_RANGE = range(UNSIGNED_UPPER_BOUND)
+SIGNED_RANGE = range(SIGNED_LOWER_BOUND, SIGNED_UPPER_BOUND)
+
+
+def unsigned_to_signed(u: int) -> int:
+    return u + SIGNED_LOWER_BOUND
+
+
+def test_bitwidth_2_values():
+    """
+    Above calculations are correct for bitwidth 2.
+    """
+    assert len(UNSIGNED_RANGE) == len(SIGNED_RANGE)
+    for i, u in zip(SIGNED_RANGE, UNSIGNED_RANGE):
+        assert i == unsigned_to_signed(u)
```

Comment:
```suggestion
    assert list(SIGNED_RANGE) == list(map(unsigned_to_signed, UNSIGNED_RANGE))
```

Reply:
this is less useful for unit tests, a list of mismatched elements is harder to compare than a specific failing element in a loop

Reply:
Hmm I remember it being less helpful than this, this looks good, I'll change it, thank you